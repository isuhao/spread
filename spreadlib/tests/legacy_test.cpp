#include "common.cpp"

bf::path mydir = "_legacy/";
bf::path chandir = mydir/"channels/test/";
bf::path indir = "input_data/legacy_test/";

int main()
{
  bf::remove_all(mydir);
  bf::create_directories(chandir);
  bf::copy_file(indir/"packs.json", chandir/"packs.json");

  {
    cout << "Setting test2 with no dir file\n";
    SpreadLib lib(mydir.string(), (mydir/"tmp").string());
    lib.setLegacyPack("test", "test2", (mydir/"out").string());
    printDir(mydir.string(), true);
  }
  cout << "Restarting spread\n";
  SpreadLib lib(mydir.string(), (mydir/"tmp").string());
  cout << "Setting test1 WITH dir file\n";
  lib.cacheFile((indir/"test.dir").string());
  lib.setLegacyPack("test", "test1", (mydir/"out").string());
  printDir(mydir.string(), true);
  cout << "Calling again\n";
  lib.setLegacyPack("test", "test1", (mydir/"out").string());
  printDir(mydir.string(), true);
  return 0;
}
