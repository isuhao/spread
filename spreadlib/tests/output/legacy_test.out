Setting test2 with no dir file

Directory: _legacy/
00                                                 cache/
00                                                 cache/x9/
00                                                 channels/
00                                                 channels/test/
Qu49nKk1sCGlGXeDv2IyJ-YsYXkUnjv2zXGWBIpp1M2n       channels/test/packs.json
h4TgjaLNE2Mb5TCpM882Z0c0SbE5Wj6tG-ymKNs1QwWo       installed.conf
Total 6 elements
Hash: -TaSLtGL6RXj_b3aJHce45RR3wXxFsjLZVKtrG5Djz5QAQ
Restarting spread
Setting test1 WITH dir file

Directory: _legacy/
00                                                 cache/
00                                                 cache/x9/
x9pMJPbmhKy7oMdiSWdctjtEN_B9IDQDg_fOc2X1koeNAQ     cache/x9/pMJPbmhKy7oMdiSWdctjtEN_B9IDQDg_fOc2X1koeNAQ
00                                                 channels/
00                                                 channels/test/
Qu49nKk1sCGlGXeDv2IyJ-YsYXkUnjv2zXGWBIpp1M2n       channels/test/packs.json
QhY61tbk2QvGqJIRb4cOR9Rh6oOMaOgJijuBHlQQF6xMAQ     installed.conf
h4TgjaLNE2Mb5TCpM882Z0c0SbE5Wj6tG-ymKNs1QwWo       installed.conf.old
Total 8 elements
Hash: VMVvDt8gc_czOIi8ntpeA1__YX89a-FykWdEu2pKom_rAQ
Calling again

Directory: _legacy/
00                                                 cache/
00                                                 cache/x9/
x9pMJPbmhKy7oMdiSWdctjtEN_B9IDQDg_fOc2X1koeNAQ     cache/x9/pMJPbmhKy7oMdiSWdctjtEN_B9IDQDg_fOc2X1koeNAQ
00                                                 channels/
00                                                 channels/test/
Qu49nKk1sCGlGXeDv2IyJ-YsYXkUnjv2zXGWBIpp1M2n       channels/test/packs.json
QhY61tbk2QvGqJIRb4cOR9Rh6oOMaOgJijuBHlQQF6xMAQ     installed.conf
h4TgjaLNE2Mb5TCpM882Z0c0SbE5Wj6tG-ymKNs1QwWo       installed.conf.old
Total 8 elements
Hash: VMVvDt8gc_czOIi8ntpeA1__YX89a-FykWdEu2pKom_rAQ
